### ISSUE TITLE
Nasal runtime error in dual-control

### ISSUE THREAD

**Issue by MariuszXC** [HHS81/c182s#482]
Hello, after yesterday's MP event my fg log is full of error messages as below.

c182s$ git show-branch --sha1-name
[4e23df9] ELT: fixed bug: When crashed and repairing, ELT message was replayed again

flightgear$ git show-branch --sha1-name next 
[b216658ba] AI improved entry into parking (heading and postition)

```
 3446.67 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1457: Nasal runtime error: string index 4 out of bounds (size: 2)
 3446.67 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   at /opt/fg/install/flightgear/fgdata/Nasal/mp_broadcast.nas, line 323
 3446.67 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Nasal/mp_broadcast.nas, line 379
 3446.67 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 460
 3446.67 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 437
 3446.67 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Nasal/mp_broadcast.nas, line 440
 3446.67 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 465
 3446.67 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/myHangar/Aircraft/c182s/Models/c182s.ac, line 198
 3446.93 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1457: Nasal runtime error: string index 4 out of bounds (size: 2)
 3446.93 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   at /opt/fg/install/flightgear/fgdata/Nasal/mp_broadcast.nas, line 323
 3446.93 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Nasal/mp_broadcast.nas, line 379
 3446.93 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 460
 3446.93 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 437
 3446.93 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Nasal/mp_broadcast.nas, line 440
 3446.93 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 465
 3446.93 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/myHangar/Aircraft/c182s/Models/c182s.ac, line 198
 3447.22 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1457: Nasal runtime error: string index 4 out of bounds (size: 2)
 3447.22 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   at /opt/fg/install/flightgear/fgdata/Nasal/mp_broadcast.nas, line 323
 3447.22 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Nasal/mp_broadcast.nas, line 379
 3447.22 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 460
 3447.22 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 437
 3447.22 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Nasal/mp_broadcast.nas, line 440
 3447.22 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 465
 3447.22 [ALRT]:nasal     /opt/fg/flightgear/src/Scripting/NasalSys.cxx:1473:   called from: /opt/fg/myHangar/Aircraft/c182s/Models/c182s.ac, line 198
```

**Comment by MariuszXC** [comment]
Hello, 

I did not want to imply that this occurred at a certain commit id. I Was only reporting what code I had (both c182 and fg) running at the time I noticed it happening.
Would it be possible to add a check if a missing property is nil and skip the attempt at updating, or is it not worth the hassle?

**Comment by hbeni** [comment]
Another instance, I *could* reproduce:

```
  160.76 [ALRT]:nasal      Nasal runtime error: non-objects have no members
  160.76 [ALRT]:nasal        at /home/beni/Dokumente/FGFS/Aircraft/c182s/Nasal/c182s-aircraft-dual-control.nas, line 420
  160.76 [ALRT]:nasal        called from: /usr/share/games/flightgear-build/install/flightgear/fgdata/Aircraft/Generic/DualControl/dual-control-tools.nas, line 340
  160.76 [ALRT]:nasal        called from: /usr/share/games/flightgear-build/install/flightgear/fgdata/Aircraft/Generic/DualControl/copilot-dual-control.nas, line 230
  160.76 [ALRT]:nasal        called from: /usr/share/games/flightgear-build/install/flightgear/fgdata/Aircraft/Generic/DualControl/copilot-dual-control.nas, line 246
  160.76 [ALRT]:nasal        called from: /usr/share/games/flightgear-build/install/flightgear/fgdata/Aircraft/Generic/DualControl/copilot-dual-control.nas, line 247
```

This was triggered when tuning NAV1 to a VOR

**Comment by hbeni** [comment]
fixed by [5498f25](https://github.com/HHS81/c182s/commit/5498f259efd8c7e5f0c468f8a4d02d0136cd061b)
